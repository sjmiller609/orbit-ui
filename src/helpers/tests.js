import { BrowserRouter } from 'react-router-dom'
import { shape } from 'prop-types'
import { mount } from 'enzyme'

// Instantiate router context
const router = {
  history: new BrowserRouter().history,
  route: {
    location: {},
    match: {},
  },
}

const createContext = () => ({
  context: { router },
  childContextTypes: { router: shape({}) },
})

// wrap with router context - to mount RR links correctly
export function mountWrap(node) {
  return mount(node, createContext())
}
