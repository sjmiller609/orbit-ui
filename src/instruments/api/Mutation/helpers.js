const has = (str, find) => ~str.toLowerCase().indexOf(find.toLowerCase())

export const formErrors = ({ error, errors }) => {
  if (!error) return

  let err = JSON.stringify(error).toLowerCase()
  return errors.find(e => {
    if (e.key && has(err, e.key)) return e
    else if (e.allKeys && e.allKeys.every(k => has(err, k))) return e
    else if (e.someKeys && e.someKeys.some(k => has(err, k))) return e
  })
}
