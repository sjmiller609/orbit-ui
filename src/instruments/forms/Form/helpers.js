// faltten and stringify nested keys
export const unpack = (obj, ns, hash) => {
  if (!obj) return
  return Object.entries(obj).reduce((acc, [key, value]) => {
    const keypath = ns ? `${ns}.${key}` : key
    if (value && typeof value === 'object') {
      return unpack(value, keypath, acc)
    }
    acc[keypath] = value
    return acc
  }, hash || {})
}

const roll = (keys, obj, value) => {
  const key = keys[0]
  // case 1 - not obj
  if (keys.length === 1) {
    obj[key] = value
    return obj
  }
  // case 2 - doesn't exist
  if (!obj[key]) {
    obj[key] = isNaN(keys[1]) ? {} : []
  }
  // case 3 - exists
  obj[key] = roll(
    keys.slice(1),
    Array.isArray(obj[key]) ? [...obj[key]] : { ...obj[key] },
    value
  )

  return obj
}

// convert back into obj
export const pack = (obj, type = {}) => {
  return Object.entries(obj).reduce((acc, [key, value]) => {
    const keys = key.split('.')

    return roll(keys, acc, value)
  }, type)
}

export const packChild = ({ name, obj }) => {
  const obj2 = {}
  let isArray
  Object.entries(obj).forEach(k => {
    if (name.length < k[0].length && k[0].indexOf(name) === 0) {
      const n = k[0].slice(name.length + 1)
      // must be true for all matching
      isArray = !isNaN(n.slice(0, n.indexOf('.')))
      obj2[n] = k[1]
    }
  })
  if (Object.keys(obj2).length) return pack(obj2, isArray ? [] : {})
  return null
}

export const removeChild = ({ name, obj = {} }) => {
  const obj2 = {}
  Object.entries(obj).forEach(k => {
    if (name.length >= k[0].length || k[0].indexOf(name) !== 0) {
      obj2[k[0]] = k[1]
    }
  })
  return obj2
}
