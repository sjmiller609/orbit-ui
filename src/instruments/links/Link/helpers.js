// check if url is external
export const externalUrl = url => {
  if (typeof url === 'object') return false
  const path = /^https?:\/\/|^\/\//i
  if (path.test(url) && !~url.indexOf(window.location.origin)) return true
  return
}
