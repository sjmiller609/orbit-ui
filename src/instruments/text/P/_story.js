import React from 'react'
import { storiesOf } from '@storybook/react'
import { P } from 'instruments'

storiesOf('Instruments|Text', module).add('P', () => (
  <P>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed tincidunt ipsum
    lacus, nec mollis arcu vulputate eget. Vestibulum est libero, finibus
    commodo dictum vel, faucibus euismod neque. Quisque ullamcorper nibh nec
    scelerisque aliquam. Proin eros dolor, pretium vel efficitur sed, dictum in
    lorem.{' '}
  </P>
))
