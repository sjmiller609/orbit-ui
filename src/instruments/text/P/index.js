'use strict'
import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import s from './styles.scss'

const P = ({ children, center, className }) => {
  return (
    <p className={classnames(s.p, center && s.center, className)}>{children}</p>
  )
}

P.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.element,
    PropTypes.array,
  ]),
  className: PropTypes.string,
  center: PropTypes.bool,
}

export default P
