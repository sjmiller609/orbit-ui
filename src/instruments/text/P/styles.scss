@import '../../../styles/mixins.scss';

.p {
  @include font($f400);
  font-size: 1.5rem;

  a {
    display: inline;
    @include font($f600);
    &:hover,
    &:active,
    &:focus {
      color: $lightGreen;
    }
    &,
    &:visited {
      color: $green;
    }
  }
}
