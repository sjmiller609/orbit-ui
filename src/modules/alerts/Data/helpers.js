import { formErrors } from 'instruments'

const errors = [
  {
    key: 'permissions',
    name: 'properties.alert_emails',
    error: 'Contact your workspace admin to upgrade your permissions.',
  },
]

export const handleError = error => formErrors({ error, errors })

export const trimError = error => error.split(':')[1].trim()
