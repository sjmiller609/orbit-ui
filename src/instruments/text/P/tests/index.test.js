import React from 'react'
import { mountWrap } from 'helpers/tests'
import P from '../index'

describe('P', () => {
  it('should be defined', () => {
    expect(P).toBeDefined()
  })

  it('should render correctly', () => {
    const wrapper = mountWrap(<P className="testClassName">Children</P>)
    expect(wrapper).toMatchSnapshot()
  })

  it('should handle center', () => {
    const wrapper = mountWrap(
      <P className="testClassName" center={true}>
        Children
      </P>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
