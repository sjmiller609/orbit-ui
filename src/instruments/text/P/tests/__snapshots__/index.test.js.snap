// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`P should handle center 1`] = `
<P
  center={true}
  className="testClassName"
>
  <p
    className="p center testClassName"
  >
    Children
  </p>
</P>
`;

exports[`P should render correctly 1`] = `
<P
  className="testClassName"
>
  <p
    className="p testClassName"
  >
    Children
  </p>
</P>
`;
